### PR TITLE
feat: add Support for New IoT V10

### PR DESCRIPTION
**Pull request by jrk94**

### CODE REVIEW

File: cmf-cli/Commands/build/IoTLibCommand.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+using System.CommandLine.NamingConventionBinder;
+using System.IO.Abstractions;
+using System.Linq;
+
+namespace Cmf.CLI.Commands.New
+{
+    /// <summary>
+    /// Generates IoT package structure
+    /// </summary>
+    [CmfCommand("lib", ParentId = "build_iot", Id = "build_iot_lib")]
```
Comment (m-s-) [inline cmf-cli/Commands/build/IoTLibCommand.cs:17]: add Description attr for help text

File: cmf-cli/Commands/new/IoTCommand.cs [hunk 2]
```diff
@@ -57,5 +77,121 @@ protected override List<string> GenerateArgs(IDirectoryInfo projectRoot, IDirect
 
             return args;
         }
+
+        /// <summary>
+        /// Execute the command
+        /// </summary>
+        /// <param name="workingDir">nearest root package</param>
+        /// <param name="version">package version</param>
+        public void Execute(IDirectoryInfo workingDir, string version, string htmlPackageLocation)
```
Comment (m-s-) [inline cmf-cli/Commands/new/IoTCommand.cs:86]: update comment

File: cmf-cli/Commands/new/IoTCommand.cs [hunk 3]
```diff
@@ -57,5 +77,121 @@ protected override List<string> GenerateArgs(IDirectoryInfo projectRoot, IDirect
...
+
+            var iotRoot = cmfPackage.GetFileInfo().Directory;
+            var iotCustomPackageWorkDir = iotCustomPackage.GetFileInfo().Directory;
+            var iotCustomPackageName = base.GeneratePackageName(iotCustomPackageWorkDir)!.Value.Item1;
+
+            var mesVersion = ExecutionContext.Instance.ProjectConfig.MESVersion;
+
+            var schematicsVersion = ngxSchematicsVersion.ToString() ?? $"@release-{mesVersion.Major}{mesVersion.Minor}{mesVersion.Build}";
+
+            Log.Debug($"Creating new web application {packageName}");
```
Comment (m-s-) [inline cmf-cli/Commands/new/IoTCommand.cs:141]: not a web application, it's a workspace for angular libraries

File: cmf-cli/Commands/new/IoTCommand.cs [hunk 4]
```diff
@@ -57,5 +77,121 @@ protected override List<string> GenerateArgs(IDirectoryInfo projectRoot, IDirect
...
+                DisplayName = "npm yeoman generator-iot",
+                Args = new string[] { "install", $"@criticalmanufacturing/generator-iot@{mesVersion.Major}{mesVersion.Minor}x", "--save-dev" },
+                WorkingDirectory = iotCustomPackageWorkDir
+            }.Exec();
+
+            #region Link To HTML Package
+
+            IDirectoryInfo htmlPackageDir = fileSystem.DirectoryInfo.New(htmlPackageLocation);
+
+            if (!htmlPackageDir.Exists) throw new CliException(string.Format(CliMessages.SomePackagesNotFound, string.Join(", ", htmlPackageLocation)));
```
Comment (m-s-) [inline cmf-cli/Commands/new/IoTCommand.cs:185]: Check for mandatory options at the beginning

File: cmf-cli/resources/template_feed/iot/from1000/IoT.Package/%iotpackages%/cmfpackage.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
+  "packageId": "<%= $CLI_PARAM_CustomPackageName %>.Packages",
+  "version": "<%= $CLI_PARAM_CustomPackageVersion %>",
+  "description": "Cmf Custom <%= $CLI_PARAM_Tenant %> <%= $CLI_PARAM_CustomPackageName %>.Packages Package",
+  "packageType": "IoT",
+  "isInstallable": true,
+  "isUniqueInstall": false,
+  "contentToPack": [
+    {
+      "source": "projects/*",
```
Comment (m-s-) [inline cmf-cli/resources/template_feed/iot/from1000/IoT.Package/%iotpackages%/cmfpackage.json:10]: you're packing the source, not the compiled angular output

File: core/Objects/ExecutionContext.cs [hunk 6]
```diff
@@ -70,6 +72,8 @@ public class ExecutionContext
         /// </summary>
         public static string EnvVarPrefix { get; set; }
 
+        public static Dictionary<string, KeyValuePair<RelatedPackage, IPackageTypeHandler>> RelatedPackages = new();
```
Comment (m-s-) [inline core/Objects/ExecutionContext.cs:75]: comments pls 🗡️

File: tests/Specs/New.cs [hunk 7]
```diff
@@ -295,6 +297,43 @@ public void IoTPackage()
...
+            Assert.True(Directory.Exists($"{packageId}/{packageFolder}/src"), "src is missing");
+            Assert.True(File.Exists($"{packageId}/{packageFolder}/angular.json"), "angular.json is missing");
+            Assert.True(File.Exists($"{packageId}/{packageFolder}/package.json"), "package.json is missing");
+
+            Assert.Equal(packageIdData, TestUtilities.GetPackageProperty("packageId", $"{packageId}/{packageFolder}/cmfpackage.json"), "Package Id does not match expected");
+            Assert.Equal(PackageType.IoT.ToString(), TestUtilities.GetPackageProperty("packageType", $"{packageId}/{packageFolder}/cmfpackage.json"), "Package Type does not match expected");
+            Assert.Equal(TestUtilities.GetPackageProperty("version", $"{packageId}/cmfpackage.json"),
+                TestUtilities.GetPackageProperty("version", $"{packageId}/{packageFolder}/cmfpackage.json"), "Version does not match expected");
+            File.ReadAllText(Path.Join(dir, packageId, "cmfpackage.json"))
+                .Should().Contain(@"CriticalManufacturing.DeploymentMetadata", "VM Dependency should not be included in root package");
```
Comment (m-s-) [inline tests/Specs/New.cs:332]: message doesn't match condition

**Review by m-s- — APPROVED** [review]

**Review by m-s- — DISMISSED** [review]

**Review by joaoopereira — APPROVED** [review]

**Pull request merged**